runtime/src/message_processor.rs

```diff
@@ -86,10 +89,15 @@ impl MessageProcessor {
             .zip(program_indices.iter())
             .enumerate()
         {
-            if invoke_context
-                .feature_set
-                .is_active(&prevent_calling_precompiles_as_programs::id())
-                && is_precompile(program_id, |id| invoke_context.feature_set.is_active(id))
+            let is_precompile =
+                is_precompile(program_id, |id| invoke_context.feature_set.is_active(id));
```

Comment:
Can you `&&` this with the check that the `prevent_calling_precompiles_as_programs` feature is active? Technically there are no "precompiles" unless that feature is active. We should also ensure that the usage of `is_precompile` below also incorporates the feature gate check.